todo/src/store/currentUser/userSlice.ts

```diff
@@ -0,0 +1,27 @@
+import { createSlice } from "@reduxjs/toolkit";
+import { RootState } from "../../store";
+import { useState } from "react";
+const initialState={
+  user: null ,userName:null, loggedIn:false
+};
+export const userSlice = createSlice({
+  name: "user",
+  initialState ,
+  
+  reducers: {
+    login: (state, action) => {
+      state.user = action.payload.data;
+      state.userName=action.payload.email;
+      state.loggedIn= true ;
+    },
+    signOut:(state)=>{
+       state.user=null;
+       state.userName=null;
+       state.loggedIn=false;
```

Comment:
Instead of repeating these. You should be using the initialState variable that you have already declared on line 4.

Reply:
I have changed it now.

 signOut:(state)=>{
     return initialState
    }